cmd/der2ascii/writer.go

```diff
@@ -355,6 +355,40 @@ func derToASCIIImpl(out *bytes.Buffer, in []byte, indent int, stopAtEOC bool) []
 					// Emit the remaining as a DER element.
 					derToASCIIImpl(out, elem.body[1:], indent+1, false) // Adds a trailing newline.
 					addLine(out, indent, "}")
+				} else if len(elem.body) >= 1 && len(elem.body) <= 5 && elem.body[0] < 8 {
+					// Convert to a b`` literal when the leading bit is valid and the
+					// number of data octets is at most 4.
+					// 
+					// The 4 octet limit is as above: X.509 incorrectly encodes things
+					// like signatures and RSA moduli as BIT STRING.
+					bits := new(strings.Builder)
+
+					// The first octet is the number of unused bits.
+					significant := 8 - elem.body[0]
+					for i, octet := range elem.body[1:] {
+						// Last octet gets some special handling.
+						isLast := i == len(elem.body) - 2
+						for j := 0; j < 8; j++ {
+							if isLast {
+								if octet == 0 && significant <= 0 {
+									break
+								}
+								if significant == 0 {
+									bits.WriteRune('|')
```

Comment:
If the BIT STRING has correct padding, let's omit the padding bits

Reply:
Not sure I follow. I think this code correctly elides any trailing padding bits that are zero? For example, the text case
```go
	{
		[]byte{0x03, 0x05, 0x01, 0x30, 0x80, 0x00, 0x00},
		"BIT_STRING { b`0000110000000001000000000000000` }\n",
	},
```
would emit
```
b`0000110000000001000000000000000|0`
```
if this was not the case.

(Maybe I should include a better test case. Thoughts?)

Reply:
Ah sorry, I missed the `break`.

Optional: Do you think it would be clearer to include all the padding bits if any are non-zero? That is, drop the `octet == 0 && significant <= 0` block and instead add `if octet == 0 { break }` right before `bits.WriteRune('|')`.

I could go either way on this, but my thinking is that, of the representations of `{0x07, 0x80}`, I'd probably order them in preference by:
1. <code>b\`1\`</code> — this captures the logical value
2. <code>b\`1|0000000\`</code> — this captures what's been encoded after you pad it
3. <code>b\`1|0\`</code> — this is... something

And so it seems, by extension, `{0x07, 0xc0}` would be better described by <code>b\`1|1000000\`</code> than <code>b\`1|1\`</code> because those zeros aren't any less of padding bits than the 1. It's just that the 1 is the last "wrong" padding bit.

But I could also be persuaded by "let's output the shortest description". Thoughts?

Reply:
Went with `` b`1|1000000` ``. That one requires the least thinking, I think. `` b`1` `` is still better than `` b`1|0000000` `` though.